main/src/com/google/refine/commands/history/ApplyOperationsCommand.java

```diff
@@ -59,26 +59,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             respondCSRFError(response);
             return;
         }
-
-        Project project = getProject(request);
-        String jsonString = request.getParameter("operations");
         try {
-            ArrayNode a = ParsingUtilities.evaluateJsonStringToArrayNode(jsonString);
-            int count = a.size();
-            for (int i = 0; i < count; i++) {
-                if (a.get(i) instanceof ObjectNode) {
-                    ObjectNode obj = (ObjectNode) a.get(i);
+            Project project = getProject(request);
+            String jsonString = request.getParameter("operations");
 
-                    reconstructOperation(project, obj);
+            try {
+                ArrayNode a = ParsingUtilities.evaluateJsonStringToArrayNode(jsonString);
+                int count = a.size();
```

Comment:
```suggestion
                ArrayNode a = ParsingUtilities.evaluateJsonStringToArrayNode(jsonString);
                if (a != null) {
                    int count = a.size();
```

It looks like this method can return null (and doesn't report errors :( ), so call to `.size()` needs to be protected

Reply:
I've simplified the whole structure, to stop using this utility method and improve exception handling.